src/main/java/org/apache/sysds/runtime/frame/data/columns/RaggedArray.java

```diff
@@ -49,207 +49,261 @@ public class RaggedArray<T> extends Array<T> {
+		return _a.getAsByteArray();
 	}
 
 	@Override
 	public ValueType getValueType() {
-		throw new NotImplementedException("Unimplemented method 'getValueType'");
+		return _a.getValueType();
 	}
 
 	@Override
 	public Pair<ValueType, Boolean> analyzeValueType() {
-		throw new NotImplementedException("Unimplemented method 'analyzeValueType'");
+		return _a.analyzeValueType();
 	}
 
 	@Override
 	public FrameArrayType getFrameArrayType() {
-		throw new NotImplementedException("Unimplemented method 'getFrameArrayType'");
+		return FrameArrayType.RAGGED;
 	}
 
 	@Override
 	public long getExactSerializedSize() {
-		throw new NotImplementedException("Unimplemented method 'getExactSerializedSize'");
+		return _a.getExactSerializedSize();
 	}
 
 	@Override
 	protected Array<Boolean> changeTypeBitSet() {
-		throw new NotImplementedException("Unimplemented method 'changeTypeBitSet'");
+		return _a.changeTypeBitSet();
 	}
 
 	@Override
 	protected Array<Boolean> changeTypeBoolean() {
-		throw new NotImplementedException("Unimplemented method 'changeTypeBoolean'");
+		return _a.changeTypeBoolean();
 	}
 
 	@Override
 	protected Array<Double> changeTypeDouble() {
-		throw new NotImplementedException("Unimplemented method 'changeTypeDouble'");
+		return _a.changeTypeDouble();
 	}
 
 	@Override
 	protected Array<Float> changeTypeFloat() {
-		throw new NotImplementedException("Unimplemented method 'changeTypeFloat'");
+		return _a.changeTypeFloat();
 	}
 
 	@Override
 	protected Array<Integer> changeTypeInteger() {
-		throw new NotImplementedException("Unimplemented method 'changeTypeInteger'");
+		return _a.changeTypeInteger();
 	}
 
 	@Override
 	protected Array<Long> changeTypeLong() {
-		throw new NotImplementedException("Unimplemented method 'changeTypeLong'");
+		return _a.changeTypeLong();
 	}
 
 	@Override
 	protected Array<String> changeTypeString() {
-		throw new NotImplementedException("Unimplemented method 'changeTypeString'");
+		return _a.changeTypeString();
 	}
 
 	@Override
 	protected Array<Character> changeTypeCharacter() {
-		throw new NotImplementedException("Unimplemented method 'changeTypeCharacter'");
+		return _a.changeTypeCharacter();
 	}
 
 	@Override
 	public void fill(String val) {
-		throw new NotImplementedException("Unimplemented method 'fill'");
+		_a.fill(val);
 	}
 
 	@Override
 	public void fill(T val) {
-		throw new NotImplementedException("Unimplemented method 'fill'");
+		_a.fill(val);
 	}
 
 	@Override
 	public boolean isShallowSerialize() {
-		throw new NotImplementedException("Unimplemented method 'isShallowSerialize'");
+		return _a.isShallowSerialize();
 	}
 
 	@Override
 	public boolean isEmpty() {
-		throw new NotImplementedException("Unimplemented method 'isEmpty'");
+		return _a.isEmpty();
 	}
 
 	@Override
 	public Array<T> select(int[] indices) {
-		throw new NotImplementedException("Unimplemented method 'select'");
+		return _a.select(Arrays.stream(indices).filter(x -> x < _size && x >= 0).toArray());
```

Comment:
no filter. also include indexes inside the ragged part, just as null.

Reply:
Why no filter? What if input is just wrong?